paimon-core/src/main/java/org/apache/paimon/io/RecordLevelExpire.java

```diff
@@ -85,7 +92,10 @@ private RecordReader<KeyValue> wrap(RecordReader<KeyValue> reader) {
                     checkArgument(
                             !kv.value().isNullAt(timeField),
                             "Time field for record-level expire should not be null.");
-                    int recordTime = kv.value().getInt(timeField);
+                    int recordTime =
+                            timeFieldType == CoreOptions.TimeFieldType.SECOND
```

Comment:
Use switch case.

Reply:
Fixed in https://github.com/apache/paimon/commit/20a186ed4c7abbf08f9b89ae33f74a5075d068dd